src/auto-evo/RunResults.cs

```diff
@@ -793,6 +797,23 @@ void OutputPopulationForPatch(Species species, Patch patch, long population)
                     builder.Append('\n');
             }
 
+            // Prepare highlight species to remove the need to check those things in every loop iteration
+            var relevantHighlightSpecies = new HashSet<Species>(highlightSpecies) ?? new HashSet<Species>();
+            relevantHighlightSpecies.RemoveWhere(s => s == null || !results.ContainsKey(s));
```

Comment:
Creating a new var (especially a copy) here is probably costlier than looping through a few absent species. I assume it isn't intended to include the whole biodiversity anyways.

Also, I'm not sure that there's any reason for species to be null in this set, so I think that should rather raise an error.

Reply:
I agree and changed the code accordingly